lib/src/screens/supa_phone_auth.dart

```diff
@@ -69,27 +75,44 @@ class _SupaPhoneAuthState extends State<SupaPhoneAuth> {
           ),
           spacer(16),
           ElevatedButton(
-            child: const Text(
-              'Sign Up',
-              style: TextStyle(fontWeight: FontWeight.bold),
+            child: Text(
+              signingIn ? 'Sign In' : 'Sign Up',
+              style: const TextStyle(fontWeight: FontWeight.bold),
             ),
             onPressed: () async {
               if (!_formKey.currentState!.validate()) {
                 return;
               }
-
-              try {
-                await supaAuth.signInUserWithPhone(_phone.text, _password.text);
-                if (!mounted) return;
-                successAlert;
-                if (mounted) {
-                  Navigator.popAndPushNamed(context, widget.redirectUrl,
-                      arguments: {"phone": _phone.text});
+              if (signingIn) {
+                try {
+                  await supaAuth.signInUserWithPhone(
+                      _phone.text, _password.text);
+                  if (!mounted) return;
+                  successAlert;
+                  if (mounted) {
+                    Navigator.popAndPushNamed(context, widget.redirectUrl,
+                        arguments: {"phone": _phone.text});
+                  }
+                } on GoTrueException catch (error) {
+                  await warningAlert(context, error.message);
+                } catch (error) {
+                  await warningAlert(context, 'Unexpected error has occured');
+                }
+              } else {
+                try {
+                  await supaAuth.createNewPhoneUser(
+                      _phone.text, _password.text);
+                  if (!mounted) return;
+                  successAlert;
```

Comment:
Same here
```suggestion
                  await successAlert(context);
```

Reply:
Putting it like that was displaying a warning when you run flutter analyze

Reply:
Yup, you can use the `mounted` variable to check if the widget is mounted and therefore the method with `context` as a parameter can be called!